server/src/pages/shotindex/view.js

```diff
@@ -110,19 +109,64 @@ class Body extends React.Component {
     );
   }
 
-  getClipType(dimensions) {
-    // an image is considered a square if it is within
-    // a squareBuffer pixels of being one
-    const squareBuffer = 50;
-    if (dimensions.x - squareBuffer > dimensions.y) {
-      return "landscape";
-    } else if (dimensions.x < dimensions.y - squareBuffer ) {
-      return "portrait";
+  onSubmitForm(e) {
+    e.preventDefault();
+    let val = ReactDOM.findDOMNode(this.refs.search).value;
+    if (val) {
+      sendEvent("search", "submit");
+    } else {
+      sendEvent("clear-search", "submit");
+    }
+    controller.onChangeSearch(val);
+  }
+
+  onChangeSearch() {
+    let val = ReactDOM.findDOMNode(this.refs.search).value;
+    this.setState({defaultSearch: val});
+    if (!val) {
+      sendEvent("clear-search", "keyboard");
+      controller.onChangeSearch(val);
+      return;
+    }
+    if (this._keyupTimeout) {
+      clearTimeout(this._keyupTimeout);
+      this._keyupTimeout = undefined;
+    }
+    if (val.length > 3) {
+      this._keyupTimeout = setTimeout(() => {
+        sendEvent("search", "timed");
+        controller.onChangeSearch(val);
+      }, 1000);
+    }
+  }
+
+  onClearSearch(e) {
+    const val = '';
+    ReactDOM.findDOMNode(this.refs.search).value = val;
+    this.setState({defaultSearch: val});
+    controller.onChangeSearch(val);
+    sendEvent("clear-search", "button");
+    return null;
+  }
+
+  componentDidUpdate() {
+    if ((this.props.defaultSearch || "") !== (this.state.defaultSearch || "")) {
+      document.body.classList.add("search-results-loading");
+    } else {
+      document.body.classList.remove("search-results-loading");
     }
-    return "square";
   }
 
-  renderShot(shot, downloadUrl) {
+}
+
+class Card extends React.Component {
+  constructor(props) {
```

Comment:
You don't need any constructor (here you are just delegating to the superclass, but that's what happens automatically when a subclass doesn't define a method)